File: src/main/java/com/dochiri/hexagonal/application/user/dto/UserProfileResult.java
package com.dochiri.hexagonal.application.user.dto;

import com.dochiri.hexagonal.domain.user.User;

public record UserProfileResult(
        String userId,
        String email,
        String name,
        String role
) {
    public static UserProfileResult from(User user) {
        return new UserProfileResult(
                user.getId().value(),
                user.getEmail().value(),
                user.getName(),
                user.getRole().name()
        );
    }
}

File: src/main/kotlin/com/dochiri/hexagonal/application/user/dto/UserProfileResult.kt
package com.dochiri.hexagonal.application.user.dto

import com.dochiri.hexagonal.domain.user.User

data class UserProfileResult(
    val userId: String,
    val email: String,
    val name: String,
    val role: String
) {
    companion object {
        fun from(user: User): UserProfileResult =
            UserProfileResult(
                user.id.value,
                user.email.value,
                user.name,
                user.role.name
            )
    }
}
